bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -216,13 +266,59 @@ public static double[] hsbToXY(HSBType hsb, Gamut gamut) {
      * @throws IllegalArgumentException when input array has wrong size or exceeds allowed value range.
      */
     public static HSBType rgbToHsb(int[] rgb) throws IllegalArgumentException {
+        if (rgb.length == 4) {
```

Comment:
Javadoc of this function to be extended, parameter rgb can have 3 or 4 values, etc.
Maybe add docs from private method for rgbw as well, as private methods are not included in Javadoc.